### PR TITLE
Added largest sum contiguous subarray code in C++

### LINKED ISSUE #724: Largest Sum Contiguous Sub-Array in C++
This is an open issue for the Largest Sum Contiguous Sub-Array algorithm implementation in C++. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by beertocode**
**Fixes #724 }**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [ ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [ ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [ ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [ ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [ ] This pull request shall only be merged once the build passes.

-   [ ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-lscsa/cpp/largest_sum_contiguous_subarray.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+The first and the only argument contains an integer array, A.
+
+Output Format:
+Return an integer representing the maximum possible sum of the contiguous subarray.
+*/
+int maxSumSubArray(const vector<int> &A) {
+    int l=A.size();
+    int curmax=A[0];
+    int ans=A[0];
+    int s=0;
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/cpp/largest_sum_contiguous_subarray.cpp:14]: Codacy found an issue: [Variable 's' is assigned a value that is never used.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6392783)

**Comment by Danushka96** [comment]
Hi @beertocode 
Can you update the main Readme.md file with the algorithm you implemented? 
Just update the file and send the commit to this PR,
Thanks!

**Comment by beertocode** [comment]
> Hi @beertocode
> Can you update the main Readme.md file with the algorithm you implemented?
> Just update the file and send the commit to this PR,
> Thanks!

Hey @Danushka96 I have updated the readme file , do have a look. 
Thanks!

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
@all-contributors please add @beertocode for code and doc

**Comment by Danushka96** [comment]
Hi @beertocode 
Thanks for your contribution,

**Pull request merged**